ui/notifications.js

```diff
@@ -17,11 +20,56 @@ module.exports = function () {
     },
 
     methods: {
+      getSameRoot: function(read) {
+        return function readable (end, cb) {
+          read(end, function(end, data) {
+            if (data && data.value.content.root) {
+              // Get all messages with the same root, but only the ones after the user's most recent post.
+              SSB.db.query(
+                and(hasRoot(data.value.content.root), type('post')),
+                toCallback((err, results) => {
+                  // Look through the results from the end backwards and look for a post from the user.
+                  // If we find one, stop passing along results, so we only have the posts since the user last replied.
+                  for (var r = results.length - 1; r >= 0; --r) {
+                    if (results[r].value.author == SSB.net.id) break
+
+                    cb(false, results[r])
+                  }
+                })
+              )
+            }
+            
+            // Place the original message back in the queue.
+            cb(end, data)
+          })
+        }
+      },
+
       render: function () {
-        SSB.db.query(
-          and(mentions(SSB.net.id)),
-          toCallback((err, results) => {
-            this.messages = results
+        var self = this
+
+        pull(
+          cat([
+            // Messages directly mentioning the user.
+            SSB.db.query(
```

Comment:
Since we are getting roots for both of these, they should be capped at 25 I think. Like [this](https://github.com/ssbc/ssb-suggest/blob/027578f6afc4e05cafffb99bbec22b2b9f639c00/lib/watch.js#L115).